tap_hubspot/hubspot.py

```diff
@@ -101,7 +163,10 @@ def streams(self, start_date: datetime, end_date: datetime):
         elif self.tap_stream_id == "companies":
             yield from self.get_companies_legacy()
         elif self.tap_stream_id == "contacts":
-            yield from self.get_contacts(start_date=start_date, end_date=end_date)
+            self.event_state["contacts_start_date"] = start_date
+            self.event_state["contacts_end_date"] = end_date
```

Comment:
I don't see why this is important. Looks like it could be part of the exit condition. Is this a key part?